Mathlib/Data/NNRat/Floor.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2024 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.Data.Rat.Floor
+
+/-!
+# Floor Function for Non-negative Rational Numbers
+
+## Summary
+
+We define the `FloorSemiring` instance on `ℚ≥0`, and relate it's operators to `NNRat.cast`.
+
+Note that we cannot talk about `Int.fract`, which currently only works for rings.
```

Comment:
Doesn't that mean we are missing `Nat.fract` with the obvious definition?

Reply:
Yeah, but adding new definitions seems out of scope for a PR that adds new instances.